### PR TITLE
Plugin for TSV files. Closes  #4226 and #2746.

### LINKED ISSUE #4226: .tsv file extension support
See https://github.com/jupyterlab/jupyterlab/pull/4192 discussion @jasongrout

### PR DESCRIPTION
**Pull request by jirivrany**

### CODE REVIEW

File: packages/csvviewer-extension/src/index.ts [hunk 1]
```diff
@@ -2,74 +2,112 @@
 // Distributed under the terms of the Modified BSD License.
 
 import {
-  ILayoutRestorer, JupyterLab, JupyterLabPlugin
+    ILayoutRestorer, JupyterLab, JupyterLabPlugin
```
Comment (sccolbert) [inline packages/csvviewer-extension/src/index.ts]: Can you get rid of these unnecessary whitespace changes? We use 2 spaces for indentation.

File: packages/filebrowser-extension/src/index.ts [hunk 2]
```diff
@@ -517,12 +517,14 @@ function createContextMenu(model: Contents.IModel | undefined, commands: Command
 /**
  * Create a launcher for a given filebrowser widget.
  */
-function createLauncher(commands: CommandRegistry, browser: FileBrowser): Promise<Launcher> {
+function createLauncher(commands: CommandRegistry, browser: FileBrowser): Promise<MainAreaWidget<Launcher>> {
```
Comment (sccolbert) [inline packages/filebrowser-extension/src/index.ts]: What do these changes have to do with tsv support?

File: packages/csvviewer-extension/src/index.ts [hunk 3]
```diff
@@ -2,74 +2,112 @@
...
 
 
+
 /**
- * Export the plugin as default.
+ * The TSV file handler extension.
  */
-export default plugin;
+const tsv: JupyterLabPlugin<void> = {
+    activate: (app: JupyterLab, restorer: ILayoutRestorer): void => {
```
Comment (sccolbert) [inline packages/csvviewer-extension/src/index.ts]: Rather than move these to inline functions, can you create two module-level functions `activateCSV` and `activateTSV`? IMHO that would be cleaner.

File: packages/filebrowser-extension/src/index.ts (hunk 2 above)
Comment (jirivrany) [reply packages/filebrowser-extension/src/index.ts]: Nothing really. I did rebase of my branch from master before push. But I don't undrestand why this commit is included?

File: packages/csvviewer-extension/src/index.ts (hunk 1 above)
Comment (jirivrany) [reply packages/csvviewer-extension/src/index.ts]: Yes, I will use 2 spaces from now and fix this file.

File: packages/csvviewer-extension/src/index.ts (hunk 3 above)
Comment (jirivrany) [reply packages/csvviewer-extension/src/index.ts]: Ok, thanks for suggestion, I will do it.

File: packages/filebrowser-extension/src/index.ts (hunk 2 above)
Comment (jirivrany) [reply packages/filebrowser-extension/src/index.ts]: I agree that it's weird. But I really did not touch this file. What should I do? I'm not sure how to revert this change. 

Here is what I did:
```
git checkout master
git pull upstream master
git checkout tsv-file-support
git pull
git rebase master
git push
```

Comment (sccolbert) [reply packages/filebrowser-extension/src/index.ts]: It may be that github is showing the diff based on a common ancestor commit, and not taking into account the rebase. Once everything else is cleaned up in this PR, I'll check it out locally and make sure the actual diff is okay.

Comment (jirivrany) [reply packages/filebrowser-extension/src/index.ts]: Thank you. Please let me know if there is anything that I can change to fix this.

Comment (jasongrout) [reply packages/filebrowser-extension/src/index.ts]: Something does seem odd. For example, 228498d63cc4a3c754506083e57713a3032b0586 in this branch matches up with af09753672f358551ae655a54a5fc30db5fc5e22 in master, but they are actually different commits.

Comment (jirivrany) [reply packages/filebrowser-extension/src/index.ts]: Yes it's weird. I think the simplest solution would be to close this pull request and start a new branch from current master.

Comment (ian-r-rose) [reply packages/filebrowser-extension/src/index.ts]: Yes, it looks like you may have rebased a series of commits that had *already* been merged to master. One thing you can try is to do an interactive rebase `git rebase -i master`, which will bring up a text editor showing the commits in this PR. You can then delete the unwanted commits from the branch (with possibly some conflict resolution). But it may be simpler to recreate the changes in a fresh place.

Thanks for the contributions, and working through these annoyances!

**Comment by jasongrout** [comment]
Fixed by #4684
